go/pkg/client/client.go

```diff
@@ -431,6 +431,10 @@ type DialParams struct {
 	// grpc.WithInsecure() is passed in. Should only be used in test code.
 	NoSecurity bool
 
+	// NoAuth is true if TLS is enabled but the client does not need to authenticate
+	// with the server.
```

Comment:
Maybe comment that NoSecurity takes precendence?

Reply:
Pushed a new commit with more explicit wording. It's difficult to make this clear without sounding overly repetetive.